project/src/main/webapp/profile-edit.js

```diff
@@ -0,0 +1,13 @@
+$(document).ready(function() {
+  fetch('/profile-update')
+      .then((response) => response.json())
```

Comment:
Can use async and await style here to avoid chaining.

Reply:
extract into an an async function then you can do 
const response = await fetch('/profile-update');
const userData = await response.json();

Reply:
Done, thanks